ui/src/features/project/settings/views/credentials/schema-validator.ts

```diff
@@ -0,0 +1,82 @@
+import { z } from 'zod';
+
+import { dnsRegex } from '@ui/features/common/utils';
+import { zodValidators } from '@ui/utils/validators';
+
+const helmChartRegex =
+  /^(oci:\/\/)?([a-z0-9]+(?:[._-][a-z0-9]+)*\/)*[a-z0-9]+(?:[._-][a-z0-9]+)*$/i;
+
+const imageNameRegex = /^([a-z0-9]+(?:[._-][a-z0-9]+)*\/)*[a-z0-9]+(?:[._-][a-z0-9]+)*$/i;
```

Comment:
@krancour for **_image_** repoURLs, is it valid to include protocol in the url (e.g., `oci://ghcr.io/akuity/kargo`) ? Or should it always omit protocol (e.g., `ghcr.io/akuity/kargo`)

@Marvin9 does this support a port? (e.g., `example.com:3000/foo/bar`)